src/serializer/ResidualFunctions.js

```diff
@@ -301,9 +327,17 @@ export class ResidualFunctions {
 
           define(instance, id, funcNode);
         }
+      };
+
+      // rewritten functions shouldn't have references fixed up, and for simplicity we
+      // emit their instances in a naive way
+      if (rewrittenBody) naiveProcessInstances(additionalFunctionInstances, t.blockStatement(rewrittenBody), true);
```

Comment:
Perhaps have two separate versions of naiveProcessInstances. There will be small amount of duplicated code, but the intent should be much more discoverable. The name also needs some work.

Reply:
I think @yinghuitan also wanted to refactor this huge function, and pull out helper functions. Might be worth checking if him if came up with a competing.